### PR TITLE
Fix Caching for Glue Metastore

### PR DESCRIPTION
**Pull request by gs-asrivastav**
- Remove unused imports.
- Rebase
- Fix Imports/Checkstyles

Caching across transactions and enabling in-memory cache for Gluestore.
 [Issue #1487 ](https://github.com/prestosql/presto/issues/1487)

In continuation to: [PR #1495 ](https://github.com/prestosql/presto/pull/1495)

### CODE REVIEW

File: presto-hive/src/main/java/io/prestosql/plugin/hive/HiveMetadataFactory.java [hunk 1]
```diff
@@ -143,7 +138,7 @@ public HiveMetadata get()
     {
         SemiTransactionalHiveMetastore metastore = new SemiTransactionalHiveMetastore(
                 hdfsEnvironment,
-                CachingHiveMetastore.memoizeMetastore(this.metastore, perTransactionCacheMaximumSize), // per-transaction cache
```
Comment (Praveen2112) [inline presto-hive/src/main/java/io/prestosql/plugin/hive/HiveMetadataFactory.java]: Why do we need to remove this per-transaction cache ?

Comment (gs-asrivastav) [reply presto-hive/src/main/java/io/prestosql/plugin/hive/HiveMetadataFactory.java]: @Praveen2112 Because this.metastore used to be evaluated as GlueHiveMetastore and everytime a new cache map was being created because of memoizeMetastore so it never read from cache, it always used to boil down to: io.prestosql.plugin.hive.metastore.glue.GlueHiveMetastore#getTable

Comment (Praveen2112) [reply presto-hive/src/main/java/io/prestosql/plugin/hive/HiveMetadataFactory.java]: @gs-asrivastav But after we bind `CachingHiveMetastore` to `GlueHiveMetastore` it should point to `CachingHiveMetastore`. And the `CachingHiveMetastore` which we get from  `memoizeMetastore` will now acts as second layer of cache for our existing `Metastore`.  IIRC this second layer of cache is used to ensure consistent state for tables involved in that transaction are maintained. (In case of a non caching metastore which happens if we didn't set this config `hive.metastore-cache-ttl` or set it to 0ms)

File: presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/glue/GlueMetastoreModule.java [hunk 2]
```diff
@@ -38,23 +40,31 @@ protected void setup(Binder binder)
 
         if (buildConfigObject(HiveConfig.class).getRecordingPath() != null) {
             binder.bind(HiveMetastore.class)
-                    .annotatedWith(ForRecordingHiveMetastore.class)
-                    .to(GlueHiveMetastore.class)
-                    .in(Scopes.SINGLETON);
+                .annotatedWith(ForRecordingHiveMetastore.class)
+                .to(GlueHiveMetastore.class)
+                .in(Scopes.SINGLETON);
```
Comment (ebyhr) [inline presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/glue/GlueMetastoreModule.java]: Let's revert this change since the original indent is right. Also, we should separate such commits to correspond with commit message when needed. You can install codestyle from here. https://github.com/airlift/codestyle

File: presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/glue/GlueMetastoreModule.java [hunk 3]
```diff
@@ -38,23 +40,31 @@ protected void setup(Binder binder)
...
             binder.bind(GlueHiveMetastore.class).in(Scopes.SINGLETON);
             newExporter(binder).export(GlueHiveMetastore.class).withGeneratedName();
 
-            binder.bind(HiveMetastore.class).to(RecordingHiveMetastore.class).in(Scopes.SINGLETON);
+            binder.bind(HiveMetastore.class)
+                .annotatedWith(ForCachingHiveMetastore.class)
+                .to(RecordingHiveMetastore.class)
+                .in(Scopes.SINGLETON);
```
Comment (ebyhr) [inline presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/glue/GlueMetastoreModule.java]: nit: Right indent is 
```java
            binder.bind(HiveMetastore.class)
                    .annotatedWith(ForCachingHiveMetastore.class)
                    .to(RecordingHiveMetastore.class)
                    .in(Scopes.SINGLETON);
```

File: presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/glue/GlueMetastoreModule.java [hunk 4]
```diff
@@ -38,23 +40,31 @@ protected void setup(Binder binder)
...
         }
         else {
-            binder.bind(HiveMetastore.class).to(GlueHiveMetastore.class).in(Scopes.SINGLETON);
+            binder.bind(HiveMetastore.class)
+                .annotatedWith(ForCachingHiveMetastore.class)
+                .to(GlueHiveMetastore.class)
+                .in(Scopes.SINGLETON);
             newExporter(binder).export(HiveMetastore.class)
-                    .as(generator -> generator.generatedNameOf(GlueHiveMetastore.class));
+                .as(generator -> generator.generatedNameOf(GlueHiveMetastore.class));
```
Comment (ebyhr) [inline presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/glue/GlueMetastoreModule.java]: nit: Let's revert this change

File: presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/glue/GlueMetastoreModule.java [hunk 5]
```diff
@@ -38,23 +40,31 @@ protected void setup(Binder binder)
...
 
             Multibinder<Procedure> procedures = newSetBinder(binder, Procedure.class);
             procedures.addBinding().toProvider(WriteHiveMetastoreRecordingProcedure.class).in(Scopes.SINGLETON);
         }
         else {
-            binder.bind(HiveMetastore.class).to(GlueHiveMetastore.class).in(Scopes.SINGLETON);
+            binder.bind(HiveMetastore.class)
+                .annotatedWith(ForCachingHiveMetastore.class)
+                .to(GlueHiveMetastore.class)
+                .in(Scopes.SINGLETON);
```
Comment (ebyhr) [inline presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/glue/GlueMetastoreModule.java]: nit: Right indent is 
```java
            binder.bind(HiveMetastore.class)
                    .annotatedWith(ForCachingHiveMetastore.class)
                    .to(GlueHiveMetastore.class)
                    .in(Scopes.SINGLETON);
```

File: presto-hive/src/main/java/io/prestosql/plugin/hive/HiveMetadataFactory.java (hunk 1 above)
Comment (electrum) [reply presto-hive/src/main/java/io/prestosql/plugin/hive/HiveMetadataFactory.java]: @Praveen2112 is correct. This is used for both consistency and performance for multiple calls for the same data within the same transaction (or query). The engine may ask for the same data multiple times (perhaps across different metadata calls), and we don't want to go to the metastore for each one (especially because the results could change).

Comment (gs-asrivastav) [reply presto-hive/src/main/java/io/prestosql/plugin/hive/HiveMetadataFactory.java]: let me verify this with the binding changes made in ’GlueMetastoreModule‘, I’ll revert this change if gluestore is not called everytime for fetching table metadata.

Comment (gs-asrivastav) [reply presto-hive/src/main/java/io/prestosql/plugin/hive/HiveMetadataFactory.java]: @Praveen2112 @electrum reverted this change.

File: presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/glue/GlueMetastoreModule.java (hunk 5 above)
Comment (gs-asrivastav) [reply presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/glue/GlueMetastoreModule.java]: @ebyhr Added original indentation.

File: presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/glue/GlueMetastoreModule.java (hunk 4 above)
Comment (gs-asrivastav) [reply presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/glue/GlueMetastoreModule.java]: @ebyhr Added original indentation.

File: presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/glue/GlueMetastoreModule.java (hunk 3 above)
Comment (gs-asrivastav) [reply presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/glue/GlueMetastoreModule.java]: @ebyhr Added original indentation.

File: presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/glue/GlueMetastoreModule.java (hunk 2 above)
Comment (gs-asrivastav) [reply presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/glue/GlueMetastoreModule.java]: @ebyhr Added original indentation.

**Comment by gs-asrivastav** [comment]
@Praveen2112 @ebyhr
